packages/affiliates/libs/affiliates/dappmining.js

```diff
@@ -0,0 +1,254 @@
+const assert = require("assert");
+const highland = require("highland");
+const moment = require("moment");
+
+const { DecodeTransaction, DecodeLog } = require("../contracts");
+const { EmpAttributions, EmpBalances } = require("../processors");
+const { SharedAttributions, Balances } = require("../models");
+
+// Module will calculate the Dapp Mining rewards for a single EMP
+module.exports = ({ queries, empAbi, web3 }) => {
+  assert(queries, "requires queries bigquery interface");
+  assert(empAbi, "requires empAbi");
+  assert(web3, "requires web3");
+
+  const { toBN, fromWei } = web3.utils;
+
+  function getEventStream(empAddress, start, end) {
+    const stream = queries.streamLogsByContract(empAddress, start, end);
+    const decode = DecodeLog(empAbi);
+    return highland(stream).map(log => {
+      return decode(log, {
+        blockNumber: log.block_number,
+        blockTimestamp: moment(log.block_timestamp.value).valueOf(),
+        ...log
+      });
+    });
+  }
+
+  function getAttributionStream(empAddress, start, end) {
+    // We get all attributions from the trace stream. This is very similar to transactions query but
+    // transactions do not contain transactions called from other contracts! Trace does.
+    const stream = queries.streamTracesByContract(empAddress, start, end);
```

Comment:
so this is to enable sponsors who used proxies for interacting with their positions to still receive rewards, right? For example, if my sponsor account was behind a DSProxy?

Is this query much larger than the transactions call?

Reply:
I believe this will allow users to use proxies. I ran into an issue where some transactions were not being included when querying transactions. I didn't actually verify though they were being sent through a proxy. When I switched to trace, the missing transactions showed up.  I dont know how this affects tagging though, it may render it unusable anyway, might require a deeper look.

This is a much more expensive call, for instance Jan 1 2020 to Dec 2020: 
- traces: 693.5 GB vs transactions: 167.3 GB

Reply:
> I dont know how this affects tagging though, it may render it unusable anyway

What do you mean?

Reply:
Well I guess I'm not sure of how our tagging mechanism wo￼uld interact ￼with a proxy contract.

It's relying on knowing the Abi of the contract in order to decode the tag. Not sure that assumption holds when going through the proxy and following a "trace".

Reply:
@mrice32 I did some digging on this, and yes the missing transactions go through a proxy. Heres an example: https://etherscan.io/tx/0x2f23b55607432961f4659065969fbb7094ab75b086acda6b6a1b73756ea93a11

It seems like as long as you correctly tag the data param for the proxy call, I think this will be seen in the trace transaction as a normal tag to the contract. One side effect of this is that the from_address originates at the proxy address and not the EOA, but that doesnt matter as far as any calculations go.  This is what the decoded transaction looks like: 
![image](https://user-images.githubusercontent.com/4429761/103376605-1d843a00-4aab-11eb-9491-c9832e2635c9.png)